satellite/metainfo/loop.go

```diff
@@ -148,8 +148,6 @@ waitformore:
 			observers = append(observers, observer)
 		case <-timer.C:
 			break waitformore
-		case <-ctx.Done():
```

Comment:
Why? what was the problem with it?

Reply:
Yeah, this MUST be here.

Reply:
I am not sure why, but this was causing a lot of test failures. I plan to put it back in and try to figure out why after I get the tests to pass.